src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -76,76 +75,167 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
      * @param y The top left y coordinate of the new quad
      * @param data extra data to pass to the transfer
      */
-    function mintQuad(address to, uint256 size, uint256 x, uint256 y, bytes calldata data) external {
+    function mintQuad(
+        address to,
+        uint256 size,
+        uint256 x,
+        uint256 y,
+        bytes calldata data
+    ) external {
         require(to != address(0), "to is zero address");
-        require(
-            isMinter(msg.sender),
-            "Only a minter can mint"
-        );
+        require(isMinter(msg.sender), "Only a minter can mint");
         require(x % size == 0 && y % size == 0, "Invalid coordinates");
         require(x <= GRID_SIZE - size && y <= GRID_SIZE - size, "Out of bounds");
 
-        uint256 quadId;
         uint256 id = x + y * GRID_SIZE;
+        (uint256 quadId, , , ) = _getQuadInfo(size, id);
+        require(_owners[LAYER_24x24 + (x / 24) * 24 + ((y / 24) * 24) * GRID_SIZE] == 0, "Already minted");
+
+        checkOwner(size, x, y, 12);
+        for (uint256 i = 0; i < size * size; i++) {
+            uint256 _id = _idInPath(i, size, x, y);
+            require(_owners[id] == 0, "Already minted");
+            emit Transfer(address(0), to, _id);
+        }
 
+        _owners[quadId] = uint256(to);
+        _numNFTPerAddress[to] += size * size;
+
+        _checkBatchReceiverAcceptQuad(msg.sender, address(0), to, size, x, y, data);
+    }
+
+    function _getQuadInfo(uint256 size, uint256 id)
+        internal
+        pure
+        returns (
+            uint256 quadId,
+            uint256 layer,
+            uint256 parentSize,
+            uint256 childLayer
+        )
```

Comment:
This could increase the gas cost for example in  _regroupQuad function get child layer and quadId from _getQuadInfo. If we create two function as you suggest they both would be called thus increasing the gas cost.